chronos/streams/chunkstream.nim

```diff
@@ -68,161 +87,116 @@ proc setChunkSize(buffer: var openarray[byte], length: int64): int =
       i = i - 4
     buffer[c] = byte(0x0D)
     buffer[c + 1] = byte(0x0A)
-    result = c + 2
+    c + 2
 
 proc chunkedReadLoop(stream: AsyncStreamReader) {.async.} =
   var rstream = cast[ChunkedStreamReader](stream)
```

Comment:
Since this is a valid upcast, you can use a normal type coercion:

```nim
var rstream = ChunkedStreamReader(stream)
```

This way, the compiler will still check that the types are properly related. A `cast` is largely unchecked.